pkg/ucp/manifestservice/service.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifestservice
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/radius-project/radius/pkg/sdk"
+	"github.com/radius-project/radius/pkg/ucp/hosting"
+	ucpoptions "github.com/radius-project/radius/pkg/ucp/hostoptions"
+	"github.com/radius-project/radius/pkg/ucp/ucpclient"
+	"github.com/radius-project/radius/pkg/ucp/ucplog"
+)
+
+// Service implements the hosting.Service interface for registering manifests.
+type Service struct {
+	ucpConnection sdk.Connection
+	options       ucpoptions.UCPConfig
+}
+
+var _ hosting.Service = (*Service)(nil)
+
+// NewService creates a server to register manifests.
+func NewService(connection sdk.Connection, options ucpoptions.UCPConfig) *Service {
+	return &Service{
+		ucpConnection: connection,
+		options:       options,
+	}
+}
+
+// Name gets this service name.
+func (s *Service) Name() string {
+	return "manifestservice"
+}
+
+func waitForServer(ctx context.Context, host, port string, retryInterval time.Duration, timeout time.Duration) error {
+	address := net.JoinHostPort(host, port)
+	deadline := time.Now().Add(timeout)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("connection attempts canceled or timed out: %w", ctx.Err())
+		default:
+			conn, err := net.DialTimeout("tcp", address, retryInterval)
+			if err == nil {
+				conn.Close()
+				return nil
+			}
+
+			if time.Now().After(deadline) {
+				return fmt.Errorf("failed to connect to %s after %v: %w", address, timeout, err)
+			}
+
+			time.Sleep(retryInterval)
```

Comment:
Does this apply the retry interval twice? 

- First on the call to `DialTimeout`
- Second on the call to `time.Sleep`